backend/src/main/repositories/ClientAccountRepository.ts

```diff
@@ -84,6 +85,28 @@ export default class ClientAccountRepository implements CRUD {
     }
   };
 
+  public getClientsByRegex = async (email: string): Promise<ClientAccount[]> => {
+    try {
+      const operatorsAliases = {
+        like: Op.like,
+      };
+      const data = await ClientAccount.findAll({
+        limit: 5,
+        where: {
+          email: {
+            [operatorsAliases.like]: `${email}%`,
+          },
+        },
+      });
+      return Promise.resolve(data);
+    } catch (err: any) {
+      if (err instanceof BaseError) {
+        throw new Error(`${err.name}, message: ${err.message}`);
```

Comment:
I think you should `Promise.reject(new Error(.....))` instead of throw. Because of async it might not be catched. 
https://stackoverflow.com/questions/33445415/javascript-promises-reject-vs-throw#:~:text=Any%20time%20you%20are%20inside,callback%2C%20you%20must%20use%20reject%20.

Reply:
Solved